src/main/java/org/apache/fineract/api/BatchApi.java

```diff
@@ -119,6 +122,35 @@ private BatchDTO generateDetails (Batch batch) {
             }
         }
 
+        // calculating matrices for sub batches
+        AtomicReference<Long> subBatchFailed = new AtomicReference<>(0L);
+        AtomicReference<Long> subBatchCompleted = new AtomicReference<>(0L);
+        AtomicReference<Long> subBatchOngoing = new AtomicReference<>(0L);
+        AtomicReference<Long> subBatchTotal = new AtomicReference<>(0L);
+        allBatches.forEach(bt -> {
```

Comment:
does this foreach gives any advantage here in terms of performance? if we are using so many atomic reference variables for updating status, a normal loop might be better here, isn't it?